app/components/shared/forms/fields/FormSingleFileUpload.unsafe.tsx

```
@@ -31,24 +43,78 @@ export class FormSingleFileUploadComponent extends React.Component<
     isUploading: false,
   };
 
-  private onDropFile = (file: File) => {
-    this.setState({ isUploading: true });
+  private getImageDimensions = async (file: File): Promise<IImageDimensions> => {
+    const image = new Image();
+    image.src = URL.createObjectURL(file);
+
+    const onImageLoad = (image: HTMLImageElement): Promise<IImageDimensions> =>
+      Promise.race([
+        new Promise((resolve, reject) => {
+          image.onload = (e: Event) => {
+            if (e.target !== null) {
+              return resolve({
+                width: ((e.target as unknown) as IImageDimensions).width,
+                height: ((e.target as unknown) as IImageDimensions).height,
+              });
+            } else {
+              return reject(createMessage(ImageUploadMessage.IMAGE_UPLOAD_WRONG_IMAGE_TYPE));
+            }
+          };
+        }),
+        new Promise((_, reject) => {
+          image.onerror = () =>
+            reject(createMessage(ImageUploadMessage.IMAGE_UPLOAD_WRONG_IMAGE_TYPE));
```

reject should return a custom error and later in `checkImage` in catch we should check for error instance or type. By this, we can make this function more generic

---

done